main.tf

```diff
@@ -1,32 +1,75 @@
+  features_require_ami = local.need_bootstrap
+  need_ami_id          = local.features_require_ami && length(local.configured_ami_image_id) == 0
+
+  features_require_launch_template = length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami
+  generate_launch_template         = local.features_require_launch_template && length(local.configured_launch_template_name) == 0
+  use_launch_template              = local.features_require_launch_template || length(local.configured_launch_template_name) > 0
+
+  launch_template_id = local.use_launch_template ? (length(local.configured_launch_template_name) > 0 ? data.aws_launch_template.this[0].id : aws_launch_template.default[0].id) : ""
+  launch_template_version = local.use_launch_template ? (
+    length(local.configured_launch_template_version) > 0 ? local.configured_launch_template_version :
+    (
+      length(local.configured_launch_template_name) > 0 ? data.aws_launch_template.this[0].latest_version : aws_launch_template.default[0].latest_version
+    )
+  ) : ""
+
+  launch_template_ami = length(local.configured_ami_image_id) == 0 ? (local.features_require_ami ? data.aws_ami.selected[0].image_id : "") : local.configured_ami_image_id
+
+  autoscaler_enabled_tags = {
+    "k8s.io/cluster-autoscaler/${var.cluster_name}" = "owned"
+    "k8s.io/cluster-autoscaler/enabled"             = "true"
+  }
+  autoscaler_kubernetes_label_tags = {
+    for label, value in var.kubernetes_labels : format("k8s.io/cluster-autoscaler/node-template/label/%v", label) => value
+  }
+  autoscaler_kubernetes_taints_tags = {
+    for label, value in var.kubernetes_taints : format("k8s.io/cluster-autoscaler/node-template/taint/%v", label) => value
+  }
+  autoscaler_tags = merge(local.autoscaler_enabled_tags, local.autoscaler_kubernetes_label_tags, local.autoscaler_kubernetes_taints_tags)
+
+  node_tags = merge(
     module.label.tags,
     {
       "kubernetes.io/cluster/${var.cluster_name}" = "owned"
-    },
-    {
-      "k8s.io/cluster-autoscaler/${var.cluster_name}" = "owned"
-    },
-    {
-      "k8s.io/cluster-autoscaler/enabled" = "${var.enable_cluster_autoscaler}"
     }
   )
+  node_group_tags = merge(local.node_tags, var.enable_cluster_autoscaler ? local.autoscaler_tags : {})
+
   aws_policy_prefix = format("arn:%s:iam::aws:policy", join("", data.aws_partition.current.*.partition))
 
-  userdata_vars = {
-    before_cluster_joining_userdata = var.before_cluster_joining_userdata
-  }
+  get_cluster_data = local.enabled && (local.need_cluster_kubernetes_version || local.need_bootstrap)
+}
 
-  # Use a custom launch_template if one was passed as an input
-  # Otherwise, use the default in this project
-  launch_template = {
-    id             = coalesce(var.launch_template_id, aws_launch_template.default[0].id)
-    latest_version = coalesce(var.launch_template_version, aws_launch_template.default[0].latest_version)
-  }
+data "aws_eks_cluster" "this" {
```

Comment:
before, the pattern was to name a single resource `default`.
`this` would be good as well, we just have two naming conventions now.

Reply:
I would prefer to use "this" in new modules rather than default, and for backward compatibility and consistency mostly continued to use "default", but here referring to the "default cluster" just seemed confusing and wrong.